File: 2017/round_1b/steed.js
function solve(distance, horses) {
  let max_time = 0
  for (const [d1, s1] of horses) {
    max_time = Math.max(max_time, (distance - d1) / s1)
  }
  return distance / max_time
}

const readline = require('readline');
const util = require('util');

async function processLineByLine() {
  const iterator = readline.createInterface({
    input: process.stdin,
    crlfDelay: Infinity
  })[Symbol.asyncIterator]();

  const testCases = (parseInt((await iterator.next()).value, 10))
  for (let i = 1; i < testCases + 1; i++) {
    const [totalDistance, horseCount] = (await iterator.next()).value.split(' ').map(x => parseInt(x, 10));
    const horses =[];
    for (let h = 0; h < horseCount; h++) {
      const [distance, speed] = (await iterator.next()).value.split(' ').map(x => parseInt(x, 10));
      horses.push([distance, speed]);
    }
    const soln = solve(totalDistance, horses);
    console.log(`Case #${i}: ${soln}`);
  }
}

processLineByLine();

File: 2017/round_1b/steed.py
"""

"""


def solve_case(distance, horses):
    max_time = 0
    for d1, s1 in horses:
        max_time = max(max_time, (distance - d1) / s1)
    return distance / max_time


def run():
    cases = int(input())
    for i in range(1, cases + 1):
        d, h = (int(x) for x in input().split(" "))
        horses = []
        for _ in range(h):
            horses.append(tuple(int(x) for x in input().split(" ")))
        soln = solve_case(d, horses)
        print("Case #{}: {}".format(i, soln), flush=True)


if __name__ == "__main__":
    run()
